rasa/core/domain.py

```diff
@@ -229,14 +229,9 @@ def collect_slots(slot_dict):
     def collect_intent_properties(intent_list):
         intent_properties = {}
         for intent in intent_list:
-            if isinstance(intent, dict):
-                for properties in intent.values():
-                    if 'use_entities' not in properties:
-                        properties['use_entities'] = True
-                intent_properties.update(intent)
-            else:
-                intent = {intent: {'use_entities': True}}
-                intent_properties.update(intent)
```

Comment:
@tmbo This is used by `from_dict`. Do you think we can savely skip providing defaults here and only do it in  `get_parsing_states`?

Reply:
mhm, I think we should put the complete properties into `intent_properties` keeping the default values in a single place.

Reply:
The reason I removed this was, that you use this pattern often in the code:
`whatever.get('desired_value', defualt_value_if_not_specified)`

When I did this in `get_parsing_states` this pattern didn't work. Despite not having specified the value in the domain file, the get function didn't return the default value.

Then I found that here in `collect_intent_properties` there was already a default value set, which was different from my desired and as this code was called earlier than mine, my default value got ignored.
This was very confusing and is kind of inconsistent.

I think we should use  `whatever.get('desired_value', defualt_value_if_not_set)` everywhere or set all defaults in one place during loading the config files. Mixing both methods leads to these kinds of bugs described above.

I think setting all defaults in one place, would be the cleaner way. This would also eliminate the possibility of having bugs because of different defaults for the same properties in different places of the code.

Reply:
@mauricedoepke Let's initialize all defaults then in `collect_intent_properties` 👍

Reply:
@wochinge

collect_intent_properties is currently a static method which can but does not have to be called to obtain the necessary data for creating a Domain instance. If we initialize all defaults in `collect_intent_properties` and let the class methods rely on that, I suggest we make `collect_intent_properties` a class method and call it in the constructor. The only occasion where `collect_intent_properties` is not directly or indirectly called to create a Domain instance is in one test case which is easy to adapt.

Reply:
It's also used in `_write_domain_to_file` in `core/training/interactive.py`. Putting it in the constructor sounds like a good idea (since we also always call the constructur when we call `collect_intent_properties` ). But I'd actually keep the `classmethod` decorator since it has no `self` references.

Reply:
Sorry for the late feedback @mauricedoepke